[charts] Place ordinal ticks according to a continuous scale (#19808)
mui/mui-x: docs/data/charts/axis/axis.md

{{"demo": "TickLabelPosition.js"}}

### Ordinal tick management

Ordinal scales (`'band'` and `'point'`) display one tick per item by default.

> If you're axis is about date, you can subsample them with `timeOrdinalTicks` property.

Reviewer: ```suggestion
If your axis is about date, you can subsample them with the `timeOrdinalTicks` property.
```

It takes an array of frequencies at which ticks can be placed sorted from the largest to the smallest.
Tick are placed according to those frequencies plus the the `tickNumber`.

It can either be a subsample of built in frequencies: `'years'`, `'3-months'`, `'months'`, `'2-weeks'`, `'weeks'`, `'days'`, `'hours'`.
Or a custom definition of type `TimeTicksDefinition` which is an object made of following properties:

- `getTickNumber: (from: Date, to: Date) => number` Returns the number of ticks that will be displayed between `from` and `to` dates.

> - `isTick: (prev: Date, value: Date) => boolean` Returns `true` is a tick should be placed on `value`. For example if it's the beginning of a new month.

Reviewer: Should we also provide `next` here? 🤔 

I would assume you could want to display something before the end of the month. So you could say `in-the-last-day-of-the-month`

Reviewer: Yeah, I'm pretty sure that is the most common use case, but not everything is trading apps 😆 

My point is more "is this a common enough use-case that users would need it?"

It is not a big deal though, I would assume it to be easy to add a new parameter if needed

{{"demo": "TickLabelPosition.js"}}

### Ordinal tick management

Ordinal scales (`'band'` and `'point'`) display one tick per item by default.
If you have a date axis, you can use the `timeOrdinalTicks` property to configure which ticks to show.

It takes an array of frequencies at which ticks can be placed.
Those frequencies must be sorted from the largest to the smallest.
For example `['years', 'months', 'days']`.
Visible ticks are selected according to those frequencies plus the the `tickNumber`.

It can either be a subsample of built in frequencies: `'years'`, `'quarterly'`, `'months'`, `'biweekly'`, `'weeks'`, `'days'`, `'hours'`.

> Or a custom definition of type `TicksFrequencyDefinition`.

Reviewer 2: ```suggestion
The `timeOrdinalTicks` property can either be an implementation of the `TicksFrequencyDefinition` type or a subset of the built-in frequencies: `'years'`, `'quarterly'`, `'months'`, `'biweekly'`, `'weeks'`, `'days'`, `'hours'`.
```

> When using `timeOrdinalTicks` the property `tickPlacement` is ignored, and computation are done as if set to `'middle'`.

Reviewer 2: ```suggestion
When using `timeOrdinalTicks` the property `tickPlacement` is ignored, and the computation is done as if set to `'middle'`.
```
